### ISSUE TITLE
Add Golang lint to GitHub Actions for static Go code formatting scanning

### ISSUE THREAD

**Issue by FeynmanZhou** [notaryproject/notation#718]
### Is your feature request related to a problem?

N/A

### What solution do you propose?

[Golang lint](https://github.com/golangci/golangci-lint) is a fast Go linters runner. It runs linters in parallel, uses caching, supports yaml config. It’s used to make sure code adheres to common Go practices. It supports enable many different types of linters. 

I think we can introduce it to Notation to enable static code formatting scanning in each pull request. 

### What alternatives have you considered?

N/A

**Comment by tungbq** [comment]
Hi, I'm interested in the CI task like this. Do you need help? Can I pick up this task?

**Comment by FeynmanZhou** [comment]
Hi @tungbq , thanks for your interest! Yes, we can assign this issue to you.

**Comment by tungbq** [comment]
**Updates:**
- Research the https://github.com/golangci/golangci-lint-action official GitHub action for [golangci-lint](https://github.com/golangci/golangci-lint)
 - Add `golangci-lint-action` to the CI workflows as it's the official GitHub action for [golangci-lint](https://github.com/golangci/golangci-lint) from its authors. The action runs [golangci-lint](https://github.com/golangci/golangci-lint) and reports issues from linters.
 - Name the new workflow as `golangci-lint-action` in `.github/workflows/go-lint.yml`
 - Current behaviour:
   - Run `golangci-lint` on every PR
   - Report only new issue generated from the PR
   - Beside of PR events, there has an option in the trigger event to allow us manually run the workflow if needed (`workflow_dispatch:`)

**Created new PR #723 for this task.**

**Comment by ogulcanaydogan** [comment]
Hi! I noticed this issue has been open for a while and the previous PR (#723) was auto-closed by the stale bot. Is @tungbq still working on this?

If not, I'd be happy to pick this up. My approach would be:
1. Add a `golangci-lint-action` GitHub Actions workflow based on PR #723's approach
2. Update to the latest `golangci-lint` version and action version
3. Ensure it runs on PRs and catches common Go issues

Let me know if this is available to work on!

**Comment by yizha1** [comment]
Hi @ogulcanaydogan, thanks for your interest in this issue. Please feel free to proceed.
